core/application/tests/userWorkflows/UserService.test.ts

```diff
@@ -0,0 +1,93 @@
+import Repository from '@application/technicalImpl/policies/repositories/Repository'
+import { UserService } from '@application/userWorkflows/UserService'
+import { generateUniqueID } from '../../utils/ksuidGenerator'
+import { UserDTO } from '@commons/dto/UserDTO'
+import { getEmailVerificationMessage, getPasswordResetVerificationMessage } from '@application/userWorkflows/userMessages'
+
+jest.mock('../../utils/ksuidGenerator')
+jest.mock('@application/userWorkflows/userMessages')
+
+describe('UserService Tests', () => {
+  let userService: UserService
+  let mockRepository: jest.Mocked<Repository<UserDTO>>
+
+  const mockUserAttributes = {
+    email: 'ebrahim@example.com',
+    name: 'Ebrahim',
+    phone_number: '1234567890'
+  }
+
+  beforeEach(() => {
+    userService = new UserService()
+
+    mockRepository = {
+      create: jest.fn()
+    } satisfies jest.Mocked<Repository<UserDTO>>
+    ;(generateUniqueID as jest.Mock).mockReturnValue('unique-id')
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should create a new user successfully', async() => {
+    const mockUser: UserDTO = {
+      id: 'unique-id',
+      email: mockUserAttributes.email,
+      phone: mockUserAttributes.phone_number,
+      name: mockUserAttributes.name,
+      registrationDate: new Date()
+    }
+
+    mockRepository.create.mockResolvedValue(mockUser)
+
+    const result = await userService.createNewUser(mockUserAttributes, mockRepository)
+
+    expect(result).toEqual({
+      ...mockUser,
+      registrationDate: expect.any(Date)
```

Comment:
You're right Vai, Since TypeScript ensures the return type, so checking the structure of the mocked return value was redundant. I’ve removed that.

You will find with this [Commit](https://github.com/CraftsmenLtd/BloodConnect/pull/113/commits/505ab38fab65e6f5e55b98b225bcf43392d521fa)